astroquery/esa/integral/core.py

```diff
@@ -0,0 +1,1068 @@
+
+        if radius:
+            radius = esautils.get_degree_radius(radius)
+
+        query_params = {
+            'REQUEST': 'timelines',
+            "ra": ra,
+            "dec": dec,
+            "radius": radius
+        }
+
+        # Execute the request to the servlet
+        request_result = esautils.execute_servlet_request(url=conf.ISLA_SERVLET,
+                                                          tap=self.tap,
+                                                          query_params=query_params)
+        total_items = request_result['totalItems']
+        data = request_result['data']
+        fraFC = data['fraFC']
+        totEffExpo = data['totEffExpo']
+        timeline = Table({
+            "scwExpo": data["scwExpo"],
+            "scwRevs": data["scwRevs"],
+            "scwTimes": [datetime.fromtimestamp(scwTime / 1000) for scwTime in data["scwTimes"]],
+            "scwOffAxis": data["scwOffAxis"]
+        })
+
+        # Plot the timeline if required
+        if plot:
+            x = timeline['scwRevs'] if plot_revno else timeline['scwTimes']
+            x_label = 'Revolutions' if plot_revno else 'Calendar Dates'
+            if plot_distance:
+                esautils.plot_concatenated_results(x, timeline['scwExpo'] / 1000, timeline['scwOffAxis'],
+                                                   x_label, 'Effective Exposure (ks)', 'Distance (deg)',
+                                                   'Observations', x_label='Pointing (Ks)',
+                                                   y_label='Off-axis (deg)')
+            else:
+                esautils.plot_result(x, timeline['scwExpo'] / 1000,
+                                     x_label, 'Effective Exposure (ks)',
+                                     'Observations')
+
+        return {'total_items': total_items, 'fraFC': fraFC, 'totEffExpo': totEffExpo, 'timeline': timeline}
+
+    def get_epochs(self, *, target_name=None, instrument=None, band=None):
+        """Retrieve the INTEGRAL epochs associated to a target and an instrument or a band
+
+        Parameters
+        ----------
+        target_name : str, optional
+            target name to be requested, mandatory
+        instrument : str, optional
+            Possible values are in isla.instruments object
+        band : str, optional
+            Possible values are in isla.bandsobject
+
+        Returns
+        -------
+        An astropy.table object containing the available epochs
+        """
+
+        value = self.__get_instrument_or_band(instrument=instrument, band=band)
+        instrument_oid, band_oid = self.__get_oids(value)
+        if target_name:
+            query = conf.ISLA_EPOCH_TARGET_QUERY.format(target_name, instrument_oid, band_oid)
+        else:
+            query = conf.ISLA_EPOCH_QUERY.format(instrument_oid, band_oid)
+        return self.query_tap(query)
+
+    def get_long_term_timeseries(self, target_name, *, instrument=None, band=None, plot=False):
```

Comment:
I wonder if the get_* and download* methods can be merged into one where the download is just a parameter, after all they are almost identical? Would that make sense?

Reply:
They were providing different outputs, but it is true that the valid ones are retrieved using download methods. So, I have removed the original get_* methods and renamed the download_* to get_*, plus other features.